torch/csrc/api/src/optim/lbfgs.cpp

```diff
@@ -20,6 +20,10 @@ LBFGSOptions::LBFGSOptions(double learning_rate)
 Tensor LBFGS::gather_flat_grad() {
   std::vector<Tensor> views;
   for (auto& parameter : parameters_) {
+    if (!parameter.grad().defined()) {
+      continue;
+    }
+
```

Comment:
It would be awesome to add a minimal test for this fix, to show that the bug doesn't occur anymore after the fix. :D Thanks!